flask_appbuilder/babel/views.py

```diff
@@ -13,6 +14,12 @@ class LocaleView(BaseView):
     def index(self, locale):
         if locale not in self.appbuilder.bm.languages:
             abort(404, description="Locale not supported.")
+
+        if request.referrer is not None:
+            page_history = Stack(session.get("page_history", []))
+            page_history.push(request.referrer)
+            session["page_history"] = page_history.to_json()
+
         session["locale"] = locale
         refresh()
         self.update_redirect()
```

Comment:
The `LocaleView.index` is already calling `update_redirect` why updating it again? because we need to update it with the referrer?

Also would it make more sense to add the `update_redirect` to Superset?

Reply:
> The LocaleView.index is already calling update_redirect why updating it again? because we need to update it with the referrer?

Yes,  updating it with the referrer before calling `update_redirect` would ensure it redirects back to the referrer page.

> Also would it make more sense to add the update_redirect to Superset?

Surely you are right:) It is just a bit tough to add `update_redirect` in every related view, and the same case even with 
 build-in pages, not only Superset.

For example, with the URL "/login/", that is, `login` function at subclass of `class AuthView`, e.g [class AuthDBView](https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/views.py#L513),  it also does not call the `update_redirect` and would redirect to other page while changing locale.